packages/create-web/index.js

```diff
@@ -0,0 +1,215 @@
+    //
+    // The filename argument is used only for error messages.
+    let state = "normal";
+
+    const outputLines = [];
+    let lineNum = 0;
+    for (const line of inputLines) {
+        lineNum++;
+        // Update the state
+        const match = line.trim().match(magicCommentRgx);
+        if (match) {
+            const newState = match[1];
+            if (newState === "template" || newState === "python" || newState === "rust" || newState === "none" || newState === "normal") {
+                state = newState;
+                // Skip the line, because we don't want these magic comments to
+                // be included in the output
+                continue;
+            } else {
+                throw new Error(`Invalid state: '${newState}' in ${fileNameRel}:${lineNum}`);
+            }
+        }
+
+        // Process the line accordingly
+        if (state === "template") {
+            continue;
+        }
+        else if (state === "python") {
+            if (backendLanguage === "python") {
+                outputLines.push(uncommentFunc(line));
+            } else continue;
+        }
+        else if (state === "rust") {
+            if (backendLanguage === "rust") {
+                outputLines.push(uncommentFunc(line));
+            } else continue;
+        }
+        else if (state === "none") {
+            if (backendLanguage === "none") {
+                outputLines.push(uncommentFunc(line));
+            } else continue;
+        }
+        else if (state === "normal") {
+            outputLines.push(line);
+        }
+    }
+    // If we reach the end of the file while in a special state, that's an error
+    if (state !== "normal") {
+        throw new Error(`Unexpected end of file while in state '${state}' in ${fileNameRel}`);
+    }
+
+    return outputLines
+}
+
+
+function patchPackageJson(
+    packageJson, // object (parsed package.json)
+    projectName, // string,
+    backendLanguage, // "python" | "rust" | "none"
+) { // -> object
+    // Fix project name
+    packageJson.name = projectName;
+
+    // Fix scripts
+    const scripts = new Map(Object.entries(packageJson.scripts)); // Map<string, string>
+    const prefix = `@@${backendLanguage} `;  // Note the space at the end
+    // Override any entries that start with the prefix
+    const entriesToOverride = [...scripts.entries()]
+        .filter(([key, _]) => key.startsWith(prefix))
+        .map(([key, val]) => [key.slice(prefix.length), val]);
+    for (const [key, val] of entriesToOverride) {
+        scripts.set(key, val);
+    }
+    // Remove all empty commands and any entries that still start with @@
+    for (const [key, val] of scripts.entries()) {
+        if (val === "" || key.startsWith("@@")) {
+            scripts.delete(key);
+        }
+    }
+    packageJson.scripts = Object.fromEntries(scripts);
+
+    // Fix dependencies. I don't know any way to not hardcode this
+    packageJson.dependencies["@uatp/components"] = "0.0.1";
```

Comment:
Obtain latest version of @uatp/components from npm https://www.npmjs.com/package/latest-version

Reply:
No, don't do that, each version of create-web needs to be pinned to one version of components

Reply:
Create issue about this & "automatic updating" or to be precise informing the user when they can update

Reply:
See https://github.com/Urban-Analytics-Technology-Platform/web-app-template/issues/29